### PR TITLE
Enable `core.hooksPath` support

### PR DESCRIPTION
**Pull request by rexagod**
Since hooks are not pushed by git, it is recommended to enable hooks
path voluntarily to run appropriate hooks and profit from a more
convenient development process. Also added a convenience target for
`etcd` and fixed existing lint errors.

NOTE: Your git version needs to be >=2.9.0 for this to work.
* `git config --local core.hooksPath hack/hooks/`

Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

### CODE REVIEW

**Comment by rexagod** [comment]
cc @aspandey @v-harihar

File: Makefile [hunk 1]
```diff
@@ -171,6 +172,19 @@ else
 KUSTOMIZE=$(shell which kustomize)
 endif
 
+# download etcd if necessary
+# requires root privileges
+etcd:
```
Comment (naveenpaul1) [inline Makefile:177]: do we need this?

Comment (rexagod) [reply Makefile:177]: Only initially, but yes. `etcd`, along with other `KUBEBUILDER_ASSETS` are required for `kubebuilder` (`operator-sdk` dependency) to work. Specifically, it's required by the `env-test` binary to run e2e tests: https://book.kubebuilder.io/reference/envtest.html

Comment (rexagod) [reply Makefile:177]: To set this up initially, contributors only need to do `make etcd` which would install all required `kubebuilder` binaries, including `etcd`.

**Comment by rexagod** [comment]
Verified on an OSD cluster.

Screenshot from 2022-05-04 15-47-52

File: controllers/monitoring_alert.go [hunk 2]
```diff
@@ -271,6 +270,7 @@ func (r *ManagedMCGReconciler) configReceiver(dmsURL string, alertingAddressList
 				receiver.EmailConfigs[0].To = strings.Join(alertingAddressList, ", ")
 				receiver.EmailConfigs[0].HTML = string(smtpHTML)
 			} else {
+				receiver.EmailConfigs = []promv1a1.EmailConfig{}
```
Comment (naveenpaul1) [inline controllers/monitoring_alert.go:273]: do we need to initialize EmailConfigs here?

Comment (rexagod) [reply controllers/monitoring_alert.go:273]: This was here originally, PTAL https://github.com/red-hat-storage/mcg-osd-deployer/pull/18/files#diff-a53791a04f3985464c97e8af8a4c8e7407c0cd6397f0dd855cd1b60a340b156fL268.

Comment (naveenpaul1) [reply controllers/monitoring_alert.go:273]: @rexagod Please check the following code is making use of that code, if not please remove it

Comment (rexagod) [reply controllers/monitoring_alert.go]: Removed and verified. Reconciliation happens without any errors.

Screenshot from 2022-05-04 17-12-56

**Comment by naveenpaul1** [comment]
//LGTM

**Pull request merged**